client/src/layers/UILayers/ToolTipWidget.vue

```diff
@@ -0,0 +1,67 @@
+<script lang="ts">
+import {
+  defineComponent, PropType, Ref,
+} from '@vue/composition-api';
+import { StateStyles } from 'vue-media-annotator/use/useStyling';
+
+export default defineComponent({
+  name: 'ToolTipWidget',
+  props: {
+    color: {
+      type: Function as PropType<(name: string) => string>,
+      required: true,
+    },
+    stateStyling: {
+      type: Object as PropType<StateStyles>,
+      required: true,
+    },
+    textSettings: {
+      type: Function as PropType<(name: string) => { showLabel: boolean; showConfidence: boolean }>,
+      default: () => ({ showLabel: true, showConfidence: true }),
+    },
+    dataList: {
+      type: Object as PropType<Ref<[string, number, number][]>>,
+      default: () => [],
+    },
+    selected: {
+      type: Object as PropType<Ref<number|null>>,
```

Comment:
`PropType<Ref<...>>` is a bit of a hack.  Props shouldn't be refs, they should be normal values.  

I understand that because this component is being mounted as the root of the application, there's no parent to drive changes in its props, but it took me a little bit of code reading to figure that out.

Could the UILayer class have a description comment block to describe this design choice, how it works, and what the special requirements of its widget components are?